test/modules/standard/List/memory.good

```diff
@@ -0,0 +1,3 @@
+0
+704
+584
```

Comment:
Hi @sagar-sam : I was wondering why the final memory was not '0' since the List's memory ought to be completely reclaimed and, using printMemAllocsByType(), found that it's due to the I/O that's done within the routine (which is ultimately cleaned up before the program exits, but unfortunately not before your final memoryUsed() call).

For that reason, would you please change this to move all I/O after all of the memoryUsed() calls are made such that the .good file can show that 0 bytes are used after output() completes?  E.g., capture the initial memoryUsed() in a variable.

BTW, while the memoryUsed() call within output() is interesting, I wouldn't include it in the test because the size of the listNode class could vary from one compiler/platform to another, which would make this test non-portable.  This will also trivially remove another I/O call.

Essentially, the goal of the above changes is to have the .good file read:

````
0
0
````

Let me know if this doesn't make sense.

Meanwhile, I'm going to fire off some testing on this branch to make sure that nothing breaks and will also see what impact it has on memory leaks on a subset of the tests.